src/org/sosy_lab/java_smt/solvers/bitwuzla/BitwuzlaTheoremProver.java

```diff
@@ -38,7 +39,10 @@ class BitwuzlaTheoremProver extends AbstractProverWithAllSat<Void> implements Pr
       new Terminator() {
         @Override
         public boolean terminate() {
-          return shutdownNotifier.shouldShutdown(); // shutdownNotifer is defined in the superclass
+          return proverShutdownManager
+              .getNotifier()
+              .shouldShutdown(); // shutdownNotifer is defined in the
+          // superclass
```

Comment:
The comment is formatted ugly, and it might not be understandable, please rephrase.

Is `getNotifier()` a "fast" method call or does it construct any objects? The callback is expected to be executed quite often from native solver code. We could store the notifier directly within the class and just reuse it.

Addtiionally, we already have a method for getting the manager. Why do we then use direct access here? Please use the getter.